### PR TITLE
Feature/6/ui mode info implementation

### PR DESCRIPTION
**Pull request by chfsx**
This is the implementation of #2328. There is currently a failing test since I have no idea how to write rendering tests for components which need other components (a glyph) in this case during rendering. any hint?

here's a screenshot of the implementation:

Bildschirmfoto vom 2019-11-28 um 12 35 18

Please note that I'm using the close-glyph and not the close-button since close-buttons do not accept actions. the rendering of the close-glyph seems broken to me, but this is not in relation of this PR.

Mobile:

Bildschirmfoto vom 2019-11-28 um 12 40 45

Please not here that the overlaying header is not related to this PR and seems to be a Bug in trunk currently

### CODE REVIEW

**Comment by Amstutz** [comment]
Hi @chfsx 

Thanks a lot I look into it ASAP. Concerning the tests, I usually do the following, e.g. see Notification Slates:

```
    public function setUp() : void
    {
        $this->sig_gen = new I\SignalGenerator();
    }

    public function getUIFactory()
    {
        $factory = new class extends NoUIFactory {
            public function button()
            {
                return new I\Button\Factory($this->sig_gen);
            }
            public function symbol() : ILIAS\UI\Component\Symbol\Factory
            {
                return new I\Symbol\Factory(
                    new I\Symbol\Icon\Factory(),
                    new I\Symbol\Glyph\Factory()
                );
            }
            public function item()
            {
                return new I\Item\Factory();
            }
            public function mainControls() : C\MainControls\Factory
            {
                return new I\MainControls\Factory($this->sig_gen,
                    new I\MainControls\Slate\Factory($this->sig_gen
                    , new \ILIAS\UI\Implementation\Component\Counter\Factory()
                    , $this->symbol()
                    )
                );
            }
            public function icon() : I\Symbol\Icon\Factory
            {
                new I\Symbol\Icon\Factory();
            }
        };
        $factory->sig_gen = $this->sig_gen;

        return $factory;
    }

```

Rendering tests are currently sub optimal, so we are open for improvements ;-).

Concering the Glyph issue, probably the wrong font is adressed. However, as you say, not your issue... If still open after merging, just hand me a ticket. But hopefully fixed by then.

**Comment by chfsx** [comment]
@Amstutz that did the trick, thanks a lot. Tests are now successful.

File: src/UI/Implementation/Component/Layout/Page/Standard.php [hunk 1]
```diff
@@ -19,7 +20,10 @@ class Standard implements Page\Standard
 {
     use ComponentHelper;
     use JavaScriptBindable;
-
+    /**
+     * @var ModeInfo
```
Comment (klees) [inline src/UI/Implementation/Component/Layout/Page/Standard.php]:
```suggestion
     * @var ModeInfo|null
```

**Review by klees — CHANGES_REQUESTED** [review]
Hi @fschmid

thanks a lot for the fast-track implementation of this really neat addition to ILIAS 6.

Please answer the following questions:
- [x] Style Changes: Note that there are many changes in lines of coded that are unrelated to this PR. This is probably to the xml Style Guide of PHP Storms, correct? Please only put lines in the PR that are clearly related to the issue at hand if they are not strictly violating our coding style guidelines.


Implement the changes or indicate, why not:
- [x] Padding/Font Size: Padding and Font size seem to be chosen somewhat arbitrarily. If possible use existing values, if not possible introduce new less variables, and comment on them, why they are so special. So we keep that info for future refactoring.
- [ ] Example: The example seems somewhat minimal. In the last JF we got the feedback, that our examples need to be a bit more relatable to our productive environment. In the light of that, can you put the mode info into a page in the example and chose Member View or similar as label, and explain, why the link is pointing nowhere. Somewhat tedious we are aware of that, but it might be helpful for viewers with a completely non-technical background.

Please implement the following changes:
- [ ] Shorthand assignments: Do not use them, especially not if they generate lines as long as the one (here)[https://github.com/ILIAS-eLearning/ILIAS/pull/2359/files#diff-1bf7a0dfe0d0af761e9b8cd484e3a951R371]

Thx a lot.

@rklees and @amstutz

**Comment by chfsx** [comment]
@klees and @Amstutz 

Thanks for the feedback, I tried to tackle most of them. Some comments or questions:

> Style Changes:

even though we now have a defined style, we don't seem to use all the right ones. I use the official PSR-2 from PHP-Storm together with the changes from ILIAS. 
But I've discarded irrelevant changes now.

> Example:

Is there a guide hoe to do that? I'm now just unfortunately the first one to have to follow this new rule. just taking a page and building the needed components is quite overkill, and doesn't really work either. A tutorial on how to do this would be great. currently it looks like this (not committed):

Bildschirmfoto 2019-11-29 um 12 28 20

> Shorthand assignments:

your link doesn't seem to be working, can you send me a new one?

**Comment by chfsx** [comment]
> Shorthand assignments: (2)

And I think we do not think about the same thing when talking about shorthand assignments, do we? :-) I think about thing like `$a += $b`

**Comment by Amstutz** [comment]
> Coding Styles

Yes, this is an issue. Stumbled upon this myself. IMO, the issue is, that a) not everybody uses PHP-Storms, and the script that can do the transform has not a completely identical output. If I find time, ill try the sort out the issue with the xml for php storms, however, not sure if I succeed in all details.

> Is there a guide hoe to do that? I'm now just unfortunately the first one to have to follow this new rule. just taking a page and building the needed components is quite overkill, and doesn't really work either. A tutorial on how to do this would be great. currently it looks like this (not committed):

This is not a rule. Just a kind question, which you might perfectly decline if you have reason to (which seems to be the case). However, can you not just import /Layout/Page/Standard/ui.php slightly adapt as needed and do something like lines 14-51. Or import other work done for the main page part, from other examples. If not just leave as is. Note that if the example is no good, sooner or later, somebody will have to do the work anyways. However, looking at your contributions in the last few days, Ill gladly pick that up if need be ;-)

> Shorthand assignments: (2) 

Should be shorthand comperisons or ternary operators (sry my bad). Link, strange it does not work anymore. New try: [Here](https://github.com/ILIAS-eLearning/ILIAS/pull/2359/files#diff-1bf7a0dfe0d0af761e9b8cd484e3a951R145)

**Comment by Amstutz** [comment]
Just skyped with @chfsx . We decided on the following points:
-> @chfsx will just extend the standard page example somewhat.
-> Ternary expression, it is none. But @chfsx will shorten the line somewhat.

@chfsx Feel free to merge as soon as done.

**Pull request merged**